src/IntlMoneyFormatter.php

```diff
@@ -42,17 +28,26 @@ public function __construct($locale, $fractionDigits)
     public function format(Money $money)
     {
         $valueBase = (string) $money->getAmount();
+        $negative = false;
+
+        if (substr($valueBase, 0, 1) === '-') {
+            $negative = true;
+            $valueBase = substr($valueBase, 1);
+        }
+
+        $fractionDigits = $this->formatter->getAttribute(\NumberFormatter::FRACTION_DIGITS);
         $valueLength = strlen($valueBase);
-        if ($valueLength > $this->fractionDigits) {
-            $subunits = substr($valueBase, 0, $valueLength - $this->fractionDigits).'.';
-            $subunits .= substr($valueBase, $valueLength - $this->fractionDigits);
+        if ($valueLength > $fractionDigits) {
+            $subunits = substr($valueBase, 0, $valueLength - $fractionDigits).'.';
+            $subunits .= substr($valueBase, $valueLength - $fractionDigits);
         } else {
-            $subunits = '0.'.str_pad('', $this->fractionDigits - $valueLength, '0').$valueBase;
+            $subunits = '0.'.str_pad('', $fractionDigits - $valueLength, '0').$valueBase;
         }
 
-        $formatter = new \NumberFormatter($this->locale, \NumberFormatter::CURRENCY);
```

Comment:
By removing custom instantiation, what if the second parameter of the cunstructor is not that constant? Will this formatter work? If not, we should document it (even if this class's constructor IMO)